modules/base/rendering/renderabletrailorbit.cpp

```diff
@@ -0,0 +1,450 @@
+                    GL_ARRAY_BUFFER,
+                    begin * sizeof(TrailVBOLayout),
+                    sizeof(TrailVBOLayout) * length,
+                    _vertexArray.data() + begin
+                );
+            };
+
+            // Only update the changed ones
+            // Since we are using a ring buffer, the number of updated needed might be
+            // bigger than our current points, which means we have to split the upload
+            // into two calls. 
+            if (report.nUpdated > 0) {
+                // deltaT is positive, so the pointer is moving backwards and update has
+                // to happen towards the front
+
+                // Starting index
+                int i = _primaryRenderInformation.first;
+                // Number of values
+                int n = report.nUpdated + 1; // +1 for the floating position
+                // Total size of the array
+                int s = _primaryRenderInformation.count;
+
+                if (i + n <= s) {
+                    // The current index is small enough to just use one upload call
+                    upload(i, n);
+                }
+                else {
+                    // The current index is too close to the wrap around part, so we need
+                    // to split the upload into two parts:
+                    // 1. from the current index to the end of the array
+                    // 2. the rest starting from the beginning of the array 
+                    int first = s - i;
+                    int second = n - first;
+                    upload(i, first);  // 1
+                    upload(0, second); // 2
+                }
+            }
+            else {
+                // deltaT is negative, so the pointer is moving forwards
+
+                // The current index
+                int i = _primaryRenderInformation.first;
+                // Number of values
+                int n = report.nUpdated + 1; // +1 for the floating position
+                // Total size of the array
+                int s = _primaryRenderInformation.count;
+
+                if (i + 1 >= n) {
+                    // The current index is big enough to fit everything into one call
+                    upload(i+1 - n, n);
+                }
+                else {
+                    // The current index is too close to the beginning of the array, so we
+                    // need to split the upload into two parts:
+                    // 1. from the beginning of the array to the current index
+                    // 2. filling the back of the array with the rest
+                    int b = n - (i + 1);
+                    upload(0, i + 1); // 1
+                    upload(s-b, b);   // 2
+                }
+            }
+        }
+    }
+
+    glEnableVertexAttribArray(0);
+    glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 0, 0);
+
+    glBindVertexArray(0);
+}
+
+RenderableTrailOrbit::UpdateReport RenderableTrailOrbit::updateTrails(
```

Comment:
The update report is only used in the parent classes render method.  The instance variables are used elsewhere, too